src/TTextEdit.cpp

```diff
@@ -1784,6 +1805,433 @@ int TTextEdit::getRowCount()
+                        .arg(tr("Lua character or code", "6th Row heading for Text analyser output, table item is either the ASCII character or the numeric code for the byte in the row about this item in the table, as displayed the thing shown can be used in a Lua string entry to reproduce this byte {this translation used 2 times}"), luaCodes);
+                isFirstRow = false;
+            } else {
+                completedRows.append(QStringLiteral("<small><table border=\"1\" style=\"margin-top:5px; margin-bottom:5px; margin-left:5px; margin-right:5px;\" width=\"100%\" cellspacing=\"2\" cellpadding=\"0\">"
+                                                    "<tr>%1</tr>"
+                                                    "<tr>%2</tr>"
+                                                    "<tr>%3</tr>"
+                                                    "<tr>%4</tr>"
+                                                    "<tr>%5</tr>"
+                                                    "<tr>%6</tr>"
+                                                    "</table></small><br>")
+                                     .arg(indexes, vals, elements, utf8Indexes, utf8Vals, luaCodes));
+            }
+            rowItems = 0;
+            indexes.clear();
+            vals.clear();
+            elements.clear();
+            utf8Indexes.clear();
+            utf8Vals.clear();
+            luaCodes.clear();
+        }
+    }
+
+    if (mpContextMenuAnalyser) {
+        if (isFirstRow) {
+            // if this is still true then we only have a short, single line of
+            // less than 16 codepoints
+            mpContextMenuAnalyser->setToolTip(QStringLiteral("<html><head/><body>%1"
+                                                             "<small><table border=\"1\" style=\"margin-top:5px; margin-bottom:5px; margin-left:5px; margin-right:5px;\" width=\"100%\" cellspacing=\"2\" cellpadding=\"0\">"
+                                                             "<tr><th>%2</th>%3</tr>"
+                                                             "<tr><th>%4</th>%5</tr>"
+                                                             "<tr><th>%6</th>%7</tr>"
+                                                             "<tr><th>%8</th>%9</tr>"
+                                                             "<tr><th>%10</th>%11</tr>"
+                                                             "<tr><th>%12</th>%13</tr>"
+                                                             "</table></small></body></html>")
+                                              .arg(completedRows)
+                                              .arg(tr("Index (UTF-16)", "1st Row heading for Text analyser output, table item is the count into the QChars/TChars that make up the text {this translation used 2 times}"), indexes)
```

Comment:
I'd recommend moving the two index rows side by side for better clarity.

Reply:
Well, at present the UTF-16 is all above the visual representation and the UTF-8 is all below and the separation is useful when the two are not in lock-step across the line e.g. consider this example I made for another PR where the differences in the indexes really shows up:
![indexing into utf16 and utf8 text example](https://user-images.githubusercontent.com/6163092/47615934-313c4c00-daae-11e8-9721-5974f4200258.png)
It is even more obvious when you stray off the Basic Multi-plain because then you start getting UTF-16 surrogate pairs and by definition at least three if not four bytes for the UTF-8 representations.